fix(starters): improve drizzle integration

sreeisalso opened the pull request:
# What is it?

- Feature / enhancement

# Description

 Updated to latest drizzle and better-sqlite3

# Checklist

- [x] My code follows the [developer guidelines of this project](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md)
- [x] I performed a self-review of my own code
- [x] I added a changeset with `pnpm change`
- [x] I made corresponding changes to the Qwik docs
- [ ] I added new tests to cover the fix / functionality

sreeisalso commented:
Why does the [Build Qwik](https://github.com/QwikDev/qwik/actions/runs/13005358799/job/36271175122?pr=7288) step is skipped in this pr, I have no expertise to check github workflow, due to which I'm not able to check the working of this pr.

gioboa commented:
> Why does the [Build Qwik](https://github.com/QwikDev/qwik/actions/runs/13005358799/job/36271175122?pr=7288) step is skipped in this pr, I have no expertise to check github workflow, due to which I'm not able to check the working of this pr.

Thanks @sreeisalso , that's because you didn't change anything about Qwik core in this PR.
is it ready for review?

sreeisalso commented:
need to test, everythig is ready

gioboa commented:
Can you add a comment with a screenshot about the working starter with Drizzle please?

sreeisalso commented:
it's not reflecting because qwik cli is not built?
> Can you add a comment with a screenshot about the working starter with Drizzle please?

![Screenshot from 2025-01-28 14-46-53](https://github.com/user-attachments/assets/3654e966-f0e5-471b-92a0-b929f52dc833)

gioboa commented:
> changes not reflecting because qwik cli is not rebuilt?

Yep, you need to use `pnpm build.local` and then link the packages.

sreeisalso commented:
> Yep, you need to use `pnpm build.local` and then link the packages. you can use [these versions](https://github.com/QwikDev/qwik/pull/7288/checks?check_run_id=36277087906) as well.

@gioboa  tried this also, but the result is same as previous screen shot

gioboa commented:
> > Yep, you need to use `pnpm build.local` and then link the packages. you can use [these versions](https://github.com/QwikDev/qwik/pull/7288/checks?check_run_id=36277087906) as well.
> 
> @gioboa tried given link also, but the result is same as previous screen shot, my machine is not able to build localy 😞

Let me check.

gioboa commented:
You can run `build.local` and then `link.dist`
Now in a new Qwik application you can run `pnpm link -g @builder.io/qwik`
now you can run `pnpm qwik add drizzle` correctly

gioboa commented:
@sreeisalso do you need help to test it?

sreeisalso commented:
> @sreeisalso do you need help to test it?

I will try today once back to system

sreeisalso commented:
I'm getting following error while building


```bash

⚛️  supabase-auth-helpers-qwik
packages/docs/src/repl/worker/repl-dependencies.ts:51:3 - error TS2740: Type '{ isServer: true; isBrowser: false; isDev: false; }' is missing the following properties from type 'typeof import("/home/sreeisalso/workspace/github/qwik/packages/qwik/dist/core")': getLocale, useServerData, withLocale, $, and 86 more.

51   self.qwikBuild = {
     ~~~~~~~~~~~~~~


Found 1 error in packages/docs/src/repl/worker/repl-dependencies.ts:51


❌ Error: Command failed with exit code 2: tsc -p packages/docs/tsconfig.json
    at makeError (file:///home/sreeisalso/workspace/github/qwik/node_modules/.pnpm/execa@8.0.1/node_modules/execa/lib/error.js:60:11)
    at handlePromise (file:///home/sreeisalso/workspace/github/qwik/node_modules/.pnpm/execa@8.0.1/node_modules/execa/index.js:124:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async tscDocs (/home/sreeisalso/workspace/github/qwik/scripts/tsc-docs.ts:8:18)
    at async build (/home/sreeisalso/workspace/github/qwik/scripts/build.ts:147:7)
 Error: Error: Command failed with exit code 2: tsc -p packages/docs/tsconfig.json
    at makeError (file:///home/sreeisalso/workspace/github/qwik/node_modules/.pnpm/execa@8.0.1/node_modules/execa/lib/error.js:60:11)
    at handlePromise (file:///home/sreeisalso/workspace/github/qwik/node_modules/.pnpm/execa@8.0.1/node_modules/execa/index.js:124:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async tscDocs (/home/sreeisalso/workspace/github/qwik/scripts/tsc-docs.ts:8:18)
    at async build (/home/sreeisalso/workspace/github/qwik/scripts/build.ts:147:7)
    at panic (/home/sreeisalso/workspace/github/qwik/scripts/util.ts:287:33)
    at build (/home/sreeisalso/workspace/github/qwik/scripts/build.ts:192:5)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
 ELIFECYCLE  Command failed with exit code 1.
```

sreeisalso commented:
tried
``` bash
pnpm build.core
pnpm link.dist

for test 
pnpm link -g @builder.io/qwik
pnpm i
```
after this it is working, please find the below logs
```bash
pnpm qwik add drizzle

> my-qwik-empty-starter@ qwik /home/sreeisalso/workspace/tmp/qwik-app
> qwik "add" "drizzle"


      ............
    .::: :--------:.
   .::::  .:-------:.
  .:::::.   .:-------.
  ::::::.     .:------.
 ::::::.        :-----:
 ::::::.       .:-----.
  :::::::.     .-----.
   ::::::::..   ---:.
    .:::::::::. :-:.
     ..::::::::::::
             ...::::
     

┌  🦋  Add Integration  drizzle
│
◇  👻  Ready?  Add drizzle to your app?
│
│  🐬 Modify
│     - package.json
│
│  🌟 Create
│     - drizzle.config.ts
│     - drizzle/schema.ts
│     - drizzle/migrations/.gitkeep
│     - drizzle/db/.gitkeep
│     - src/routes/users/index.tsx
│     - src/routes/users/[userId]/index.tsx
│     - src/routes/create/index.tsx
│
│  💾 Install pnpm dependencies:
│     - @types/better-sqlite3 ^7.6.12
│     - drizzle-kit ^0.30.3
│     - better-sqlite3 ^11.8.1
│     - drizzle-orm ^0.39.0
│
│  📜 New pnpm scripts:
│     - pnpm drizzle:generate
│     - pnpm drizzle:migrate
│     - pnpm drizzle:push
│     - pnpm drizzle:pull
│     - pnpm drizzle:check
│     - pnpm drizzle:up
│     - pnpm drizzle:studio
│     - pnpm postinstall
│
◆  Ready to apply the drizzle updates to your app?
│  ● Yes looks good, finish update!
│  ○ Nope, cancel update
└

```
result after successfull addition
```bash
◇  App updated
│
◇  New scripts added ───────╮
│                           │
│  - pnpm drizzle:generate  │
│  - pnpm drizzle:migrate   │
│  - pnpm drizzle:push      │
│  - pnpm drizzle:pull      │
│  - pnpm drizzle:check     │
│  - pnpm drizzle:up        │
│  - pnpm drizzle:studio    │
│  - pnpm postinstall       │
│                           │
├───────────────────────────╯
│
◇  🟣  Next Steps  ──────────────────────────────────────────────────────────────────────────╮
│                                                                                            │
│   Drizzle was installed with a simple DB schema and some demo routes,                      │
│                                                                                            │
│   better-sqlite3 was configured by default, but in production                              │
│   you'll want to use Postgres or MySQL.                                                    │
│                                                                                            │
│                                                                                            │
│   Drizzle studio was also added which you can access by running `npm run drizzle:studio`,  │
│                                                                                            │
│   Check out the Drizzle docs for more info:                                                │
│     - https://orm.drizzle.team/docs/overview                                               │
│                                                                                            │
├────────────────────────────────────────────────────────────────────────────────────────────╯
│
└  🦄  Success!  Added drizzle to your app
```
package.json file screenshot

![Screenshot from 2025-01-29 10-54-45](https://github.com/user-attachments/assets/4c61122c-c09f-497a-922f-f1159d64f35b)

gioboa reviewed (commented):
Thanks 🎉 that's great 👏
Let me try it

gioboa reviewed (approved):
Nice PR @sreeisalso Thanks :partying_face:
